contracts/teller-network/EscrowRelay.sol

```diff
@@ -140,79 +140,95 @@ contract EscrowRelay is RelayRecipient, Ownable {
+    uint256 /* gasLimit */,
+    uint256 /* nonce */,
+    bytes calldata /* approvalData */,
+    uint256 /* maxPossibleCharge */
+  ) external view returns (uint256, bytes memory)
+  {
+    bytes memory abiEncodedFunc = encodedFunction; // Call data elements cannot be accessed directly
     bytes4 fSign;
+    uint dataValue;
+
     assembly {
-      fSign := mload(add(encoded_function, add(0x20, 0)))
+      fSign := mload(add(abiEncodedFunc, add(0x20, 0)))
+      dataValue := mload(add(abiEncodedFunc, 36))
     }
 
-    if(from.balance > 600000 * gas_price) return ERROR_ENOUGH_BALANCE;
+    return (_evaluateConditionsToRelay(from, gasPrice, fSign, dataValue), "");
+  }
 
-    if(fSign == PAY_SIGNATURE || fSign == CANCEL_SIGNATURE || fSign == OPEN_CASE_SIGNATURE){
-        uint escrowId;
-        assembly {
-          escrowId := mload(add(encoded_function, 36))
-        }
+  /**
+   * @dev Evaluates if the sender conditions are valid for relaying a escrow transaction
+   * @param from Sender
+   * @param gasPrice Gas Price
+   * @param functionSignature Function Signature
+   * @param dataValue Represents the escrowId or offerId depending on the function being called
+   */
+  function _evaluateConditionsToRelay(address from, uint gasPrice, bytes4 functionSignature, uint dataValue) internal view returns (uint256) {
+    address token;
 
-        address token;
-        address payable buyer;
+    if(from.balance > 600000 * gasPrice) return ERROR_ENOUGH_BALANCE;
 
-        (buyer, , token, ) = escrow.getBasicTradeData(escrowId);
 
-        if(buyer != from) return ERROR_INVALID_BUYER;
-        if(token != snt && token != address(0)) return ERROR_INVALID_ASSET;
+    if(functionSignature == PAY_SIGNATURE || functionSignature == CANCEL_SIGNATURE || functionSignature == OPEN_CASE_SIGNATURE){
+      address payable buyer;
+      
+      (buyer, , token, ) = escrow.getBasicTradeData(dataValue);
 
-        if(fSign == CANCEL_SIGNATURE){ // Allow activity after 15min have passed
-            if((lastActivity[from] + 15 minutes) > block.timestamp) return ERROR_TRX_TOO_SOON;
-        }
-    } else if(fSign == CREATE_SIGNATURE) {
-      uint256 offerId;
-      assembly {
-          offerId := mload(add(encoded_function, 36))
-      }
+      if(buyer != from) return ERROR_INVALID_BUYER;
+      if(token != snt && token != address(0)) return ERROR_INVALID_ASSET;
 
-      address token = metadataStore.getAsset(offerId);
+      if(functionSignature == CANCEL_SIGNATURE){ // Allow activity after 15min have passed
+        if((lastActivity[from] + 15 minutes) > block.timestamp) return ERROR_TRX_TOO_SOON;
+      }
+    } else if(functionSignature == CREATE_SIGNATURE) {
+      token = metadataStore.getAsset(dataValue);
 
       if(token != snt && token != address(0)) return ERROR_INVALID_ASSET;
 
       // Allow activity after 15 min have passed
       if((lastActivity[from] + 15 minutes) > block.timestamp) return ERROR_TRX_TOO_SOON;
     }
-
     return OK;
```

Comment:
What happens if the functionSignature is for `createOffer`, but the suer has no ETH, would it say OK?

Reply:
`createOffer` would not be a valid function to be executed with the relayer. Only the public functions of EscrowRelay can be relayed due to it being a RelayRecipient... but I'll change it to return OK only inside the `if`s   in case another function of this contract were to be executed via relay and it was not intended to.

Reply:
Yeah, you're right, but I agree that we should only return OK for the wanted functions